### PR TITLE
[FEATURE] Document ModuleTemplate

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

**Comment by linawolf** [comment]
Depends on #4414

File: Documentation/ApiOverview/Backend/BackendModules/ModuleTemplate.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.. include:: /Includes.rst.txt
+.. index:: Backend modules; ModuleTemplateFactory
+.. _ModuleTemplate:
+
+==============
+ModuleTemplate
+==============
+
+Backend controllers should use :ref:`ModuleTemplateFactory:create <ModuleTemplateFactory>`
+to create instances of a :php:`\TYPO3\CMS\Backend\Template\ModuleTemplate.`.
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/ModuleTemplate.rst:10]: Remove dot behind class

File: Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController_ModuleTemplate.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        $menu = $moduleTemplate->getDocHeaderComponent()->getMenuRegistry()->makeMenu();
+        $menu = $this->addMenuItems($menu);
+        $moduleTemplate->getDocHeaderComponent()->getMenuRegistry()->addMenu($menu);
+
+        // Add menu items
+
+        return $moduleTemplate;
+    }
+    private function getLanguageService(): LanguageService
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController_ModuleTemplate.php:36]: Add space before

File: Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController_ModuleTemplate.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        $moduleTemplate->addFlashMessage('Everything is fine', 'Info', ContextualFeedbackSeverity::INFO);
+
+        // Assign variables to the template
+        $moduleTemplate->assignMultiple([
+            'variable1' => 'value 1',
+            'variable2' => 'value 2',
+        ]);
+        return $moduleTemplate->renderResponse('Backend/Some');
+    }
+
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController_ModuleTemplate.php:66]: Remove empty space

File: Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplate.rst.txt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    ..  php:method:: setModuleClass(string $moduleClass)
+        :returns: self
+
+        Optional 'class="module {moduleClass}"' on first <div> in body.
+
+        Can be helpful styling modules.
+
+        :param $moduleClass: the moduleClass
+
+    ..  php:method:: addFlashMessage(string $messageBody, string $messageTitle = '', TYPO3\CMS\Core\Type\ContextualFeedbackSeverity $severity = \TYPO3\CMS\Core\Type\ContextualFeedbackSeverity::OK, bool $storeInSession = true)
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplate.rst.txt:94]: Missing `\` in front of class

File: Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplate.rst.txt [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        Creates a message object and adds it to the FlashMessageQueue.
+
+        These messages are automatically rendered when the view is rendered.
+
+        :param $messageBody: the messageBody
+        :param $messageTitle: the messageTitle, default: ''
+        :param $severity: the severity, default: \TYPO3\CMS\Core\Type\ContextualFeedbackSeverity::OK
+        :param $storeInSession: the storeInSession, default: true
+
+    ..  php:method:: setFlashMessageQueue(TYPO3\CMS\Core\Messaging\FlashMessageQueue $flashMessageQueue)
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplate.rst.txt:106]: Missing `\` in front of class

File: Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        $moduleTemplate = $this->initializeModuleTemplate($this->request);
+
+        $moduleTemplate->assignMultiple([
+            'variable1' => 'value 1',
+            'variable2' => 'value 2',
+        ]);
+        return $moduleTemplate->renderResponse('Backend/Some');
+    }
+
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController.php:37]: Remove empty space

File: Documentation/ApiOverview/Backend/BackendModules/ModuleTemplate.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
+.. include:: /Includes.rst.txt
```
Comment (froemken) [inline Documentation/ApiOverview/Backend/BackendModules/ModuleTemplate.rst:1]: 2 spaces

**Review by froemken — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController_ModuleTemplate.php (hunk 2 above)
Comment (linawolf) [inline Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController_ModuleTemplate.php:36]:
```suggestion

    private function getLanguageService(): LanguageService
```

File: Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplate.rst.txt (hunk 4 above)
Comment (linawolf) [reply Documentation/ApiOverview/Backend/BackendModules/_ModuleTemplate.rst.txt:94]: These files are auto-generated...

**Comment by linawolf** [comment]
@froemken applied suggested changes

File: Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController.php [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+final class BackendModuleController extends ActionController
+{
+    public function __construct(
+        private readonly ModuleTemplateFactory $moduleTemplateFactory,
+        // ..
+    ) {}
+
+    protected function initializeModuleTemplate(ServerRequestInterface $request): ModuleTemplate
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/BackendModules/_BackendModuleController.php:18]: Why not override initializeAction() method? This is always called before an action. So no need to call initialiteModuleTemplate() manually in each action - which is error-prone (can be left out unintended). Just set $this->moduleTemplate and the use it in the action.

File: Documentation/ApiOverview/Backend/BackendModules/ModuleTemplate.rst [hunk 9]
```diff
@@ -0,0 +1,28 @@
+..  include:: /Includes.rst.txt
+.. index:: Backend modules; ModuleTemplateFactory
+.. _ModuleTemplate:
+
+==============
+ModuleTemplate
+==============
+
+Backend controllers should use :ref:`ModuleTemplateFactory:create <ModuleTemplateFactory>`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/BackendModules/ModuleTemplate.rst:9]:
```suggestion
Backend controllers should use :ref:`ModuleTemplateFactory::create() <ModuleTemplateFactory>`
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**